### PR TITLE
 AI Assistant: Enable inline Heading extension

### LINKED ISSUE #37385: AI Assistant: Enable Heading inline extension on production

### PR DESCRIPTION
**Pull request by dhasilva**
Fixes #37385

## Proposed changes:

* Adds an override to enable specific extensions without the feature flag, including the heading extension
* Generalizes the BlockHandler to allow extending other blocks later on

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Test on JN, Simple and P2
* Keep the blocks variation as "production" and do not enable the feature flag
* Go to the block editor and add a Heading block
* Check that the inline extension is enabled and working as expected
* Check that the other extensions (paragraph and list) are unchanged and working as expected

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/ai-assistant/index.ts [hunk 1]
```diff
@@ -26,20 +26,35 @@ export const isAiAssistantExtensionsSupportEnabled = getFeatureAvailability(
...
+} );
+
+unreleasedTransformativeExtensions.forEach( block => {
+	if ( isAiAssistantExtensionsSupportEnabled ) {
+		// Add the unreleased inline extension to the inline list...
+		EXTENDED_INLINE_BLOCKS.push( block );
+		// ...and remove it from the transformative list.
+		EXTENDED_TRANSFORMATIVE_BLOCKS = EXTENDED_TRANSFORMATIVE_BLOCKS.filter( b => b !== block );
+	}
+} );
```
Comment (lhkowalski) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/ai-assistant/index.ts]: This is really nice!

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/inline-extensions/lib/is-possible-to-extend-block.ts [hunk 2]
```diff
@@ -30,8 +31,11 @@ export function isPossibleToExtendBlock( blockName: string ): boolean {
 		return false;
 	}
 
-	// Check if AI Assistant extensions support is enabled.
-	if ( ! isAiAssistantExtensionsSupportEnabled ) {
+	// Check if AI Assistant extensions support is enabled or the block is in the released list
```
Comment (lhkowalski) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/inline-extensions/lib/is-possible-to-extend-block.ts]: Could we just rely on `EXTENDED_INLINE_BLOCKS.includes( blockName )` on this check?

Comment (dhasilva) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/inline-extensions/lib/is-possible-to-extend-block.ts]: Yes, we can. Good catch!

**Review by lhkowalski — APPROVED** [review]
I tested on JN, Simple and P2, and got the headings with inline extensions and the other blocks with the transformative extension, as expected.

Well done!

**Pull request merged**